sdk/trace/span.go

```diff
@@ -623,10 +623,7 @@ func (s *recordingSpan) dedupeAttrsFromRecord(record map[attribute.Key]int) {
 			record[a.Key] = len(unique) - 1
 		}
 	}
-	// s.attributes have element types of attribute.KeyValue. These types are
-	// not pointers and they themselves do not contain pointer fields,
-	// therefore the duplicate values do not need to be zeroed for them to be
-	// garbage collected.
+	clear(s.attributes[len(unique):]) // Erase unneeded elements to let GC collect objects
```

Comment:
The comment is technically correct but the conclusion is not. While there are no pointers, there are `string` and `interface{}` fields there:
- Strings contain pointers to their backing arrays that contain the actual bytes. Those arrays cannot be GCed if the string with the embedded (hidden!) pointer is reachable.
- `interface{}` can contain anything. Everything, except for pointers, in an interface is "boxed" - allocated on the heap and a pointer to that data is stored in the `interface{}`. If it's a pointer already, then it's not "boxed" but... it's a pointer and the destination cannot be GCed. So, in both cases we need to clear those `interface{}` fields.